compiler-core/src/analyse/name.rs

```diff
@@ -0,0 +1,116 @@
+use ecow::EcoString;
```

Comment:
None of the functions in this module check that the names are of the formats they are intended to check for. Using regexes here would be good.